### PR TITLE
secp256k1: --with-enable-recovery-module option and fix build on macOS 10.15

### PR DESCRIPTION
**Pull request by anton-trunov**
This PR adds `--with-enable-module-recovery` option superseding https://github.com/DomT4/homebrew-crypto/pull/95 and fixes the package on macOS 10.15 Catalina (see https://github.com/bitcoin-core/secp256k1/issues/674 -- `make` works fine, but the tests segfault).

### CODE REVIEW

File: Formula/secp256k1.rb [hunk 1]
```diff
@@ -6,17 +6,31 @@ class Secp256k1 < Formula
...
   depends_on "libtool" => :build
   depends_on "pkg-config" => :build
   depends_on "gmp"
 
   def install
+    args = %W[
+      prefix=#{prefix}
+      --disable-silent-rules
+    ]
+    # Workaround for https://github.com/bitcoin-core/secp256k1/issues/674
```
Comment (DomT4) [inline Formula/secp256k1.rb]: Do you know if this is definitely still an issue? I tested after upgrading to Xcode 11.2, as one of the posters in that thread also did, and it seems to work without this block now.

**Comment by DomT4** [comment]
Thanks for the PR! Happy to accept the option; just have the one question about the Catalina block.

File: Formula/secp256k1.rb (hunk 1 above)
Comment (anton-trunov) [reply Formula/secp256k1.rb]: Fixed

**Comment by anton-trunov** [comment]
Sorry, homebrew overwrote my fork deleting the history, so I had to reclone and fix it again in #100

**Pull request closed without merging**